server/plugin.go

```diff
@@ -356,32 +357,37 @@ func (p *Plugin) AddAutolinksForCloudInstance(ci *cloudInstance) error {
 		return fmt.Errorf("unable to get project keys: %w", err)
 	}
 
-	for _, proj := range plist {
-		key := proj.Key
-		err = p.AddAutolinks(key, ci.BaseURL)
-	}
-	if err != nil {
+	if err = p.AddAutolinks(plist, ci.BaseURL); err != nil {
 		return fmt.Errorf("some keys were not installed: %w", err)
 	}
 
 	return nil
 }
 
-func (p *Plugin) AddAutolinks(key, baseURL string) error {
+func (p *Plugin) AddAutolinks(projectList jira.ProjectList, baseURL string) error {
 	baseURL = strings.TrimRight(baseURL, "/")
```

Comment:
I find a bit difficult to follow the patterns, and unit test this may also be cumbersome, so I'd like to know what you think about these changes. Something like (the constant names aren't very good, I need a bit more context there :-) ):

```go
const (
    PATTERN_COMMENT_LINK_ENDPOINT = `/browse/)(?P<project_id>\w+)(-)(?P<jira_id>\d+)[?](focusedCommentId)(=)(?P<comment_id>\d+)`
    PATTERN_.... = `/browse/)(?P<project_id>\w+)(-)(?P<jira_id>\d+)`
    BASE_URL_... = strings.ReplaceAll(baseURL, ".", `\.`)
    TEMPLATE_COMMENT_LINK_ENDPOINT = `/browse/${project_id}-${jira_id}?focusedCommentId=${comment_id})`
    TEMPLATE_... = `/browse/${project_id}-${jira_id})`
)
```

Then we could use those later for the Autolink, like:

```go
installList := []autolink.Autolink{
    {
        ...
        Pattern: BASE_URL... + PATTERN_COMMENT_LINK_ENDPOINT
        ...
    }
}
```

We could use Sprintf too, but I think it seems not good for performance... (I'm learning Go as we speak).

Thank you! :-)

Reply:
It seems good and I have implemented the same